test-common/src/main/scala/org/scalajs/testcommon/RPCCore.scala

```diff
@@ -12,7 +14,7 @@ import java.util.concurrent.atomic.AtomicLong
 
 import Serializer.{serialize, deserialize}
 
-private[scalajs] abstract class RPCCore {
+private[scalajs] abstract class RPCCore[F[_]] {
```

Comment:
I have a vague feeling that this would be nicer as an abstract type member. Have you considered it? But I'm not too opinionated about this, so feel free to ignore my comment.

Reply:
I have indeed though so too but haven't tried out. I'm a bit concerned that `RunMuxRPC` becomes much worse with an abstract type member when used (because it introduces a path dependent type on the contained `RPCCore`).